mtf/metadata/tmet/common.py

```diff
@@ -0,0 +1,291 @@
+                self._parse_base_coverage(base=value, path=path + [key])
+
+    def _insert_to_test_tree(self, path_list, test):
+        actualelem = self.base_element
+        actualelem = actualelem[TESTE]
+        # sanitize testpath with test for these tests what are not fully qualified files
+        # append directory locaiton to source
+        if test.get(SOURCE) \
+                and not urlparse(test.get(SOURCE)).scheme \
+                and not test.get(SOURCE).startswith(os.pathsep):
+            test[SOURCE] = os.path.join(*(path_list + [test[SOURCE]]))
+            print_debug("source testpath extended %s" % test[SOURCE])
+        previous_item = None
+        link_previous = None
+        for item in path_list:
+            if actualelem.get(item) is None:
+                actualelem[item] = dict()
+            link_previous = actualelem
+            previous_item = item
+            actualelem = actualelem[item]
+        link_previous[previous_item] = test
+        self._insert_to_coverage(path_list, test)
+        return self.base_element
+
+    def _load_recursive(self):
+        allfiles = []
+        location = self.location
+        for root, sub_folders, files in os.walk(location):
+            if MFILENAME in files:
+                allfiles.append(os.path.join(root, MFILENAME))
+        elem_element = {}
+        if allfiles:
+            elem_element = self.load_yaml(allfiles[0])
+        if elem_element.get(SUBTYPE) == SUBTYPE_G:
+            allfiles = allfiles[1:]
+            self.base_element = elem_element
+        else:
+            self.base_element = {}
+        if not self.base_element.get(TESTC):
+            self.base_element[TESTC] = dict()
+        if not self.base_element.get(TESTE):
+            self.base_element[TESTE] = dict()
+        self._parse_base_coverage()
+        for item in allfiles:
+            self._insert_to_test_tree(os.path.dirname(item)[len(location):].split("/")[1:],
+                                      self.load_yaml(item))
+
+    def get_coverage(self):
+        return self.base_element.get(TESTC)
+
+    def get_backends(self):
+        return set([x.get(BACKEND) for x in self.get_coverage().values()])
+
+    def filter_relevancy(self, tests, envrion_description):
+        return tests
+
+    def filter_tags(self, tests, tag_list):
+        output = []
+        for test in tests:
+            test_tags = test.get(TAGS, "")
+            if logic_formula(test_tags, tag_list):
+                output.append(test)
+        return output
+
+    def add_filter(self, tags=[], relevancy={}):
+        addedfilter = {RELEVANCY: relevancy, TAGS: tags}
+        if self.filter_list[-1] is None:
+            self.filter_list = self.filter_list[:-1]
+        self.filter_list.append(addedfilter)
+
+    def apply_filters(self):
+        output = self.backend_tests()
+        for infilter in self.filter_list:
+            if infilter:
+                if infilter.get(TAGS):
```

Comment:
I've tested that and `if TAGS in infilter` caused regression -> in case `infilter.get(TAGS)` is empty, I had to move this code back, and also what you mentioned is fixed by upper `if infilter:` so that `infilter` cannot be empty.